### PR TITLE
fix: move prevent defaults to outside throttler

### PR DESCRIPTION
**Pull request by djcouto**

### CODE REVIEW

**Review by GoncaloCamaz — APPROVED** [review]

**Comment by miguelpinto98** [comment]
We don't have any test that check if preventDefault is enabled the click shouldn't go forward?

File: packages/ui-components/src/config.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// The number of milliseconds to throttle invocations to.
+export const DEFAULT_THROTTLE_WAIT = 300;
```
Comment (miguelpinto98) [inline packages/ui-components/src/config.ts:2]: 👍

**Review by miguelpinto98 — CHANGES_REQUESTED** [review]
Only to have some responses for my questions @djcouto

File: packages/ui-components/src/components/radio-button/radio-button.tsx [hunk 2]
```diff
@@ -40,13 +41,15 @@ export class KvRadioButton implements IRadioButton, IRadioButtonEvents, IAnchor
...
-		this.clickThrottler = throttle((event: MouseEvent) => {
-			if (this.preventDefault) {
-				event.preventDefault();
-			}
+		this.clickThrottler = throttle(() => this.onCheck(), DEFAULT_THROTTLE_WAIT);
+	}
+
+	onClick(event: MouseEvent) {
+		if (this.preventDefault) {
+			event.preventDefault();
```
[the remark this answers is not in the document]
Comment (djcouto) [reply packages/ui-components/src/components/radio-button/radio-button.tsx:49]: No, we prevent the default behaviour but we still need the `clickThrottler`  to be called.

File: packages/ui-components/src/components/tree-item/test/tree-item.spec.tsx [hunk 3]
```diff
@@ -76,4 +76,31 @@ describe('Tree Item (unit tests)', () => {
 			expect(component.counter).toEqual(32);
 		});
 	});
+
+	describe('when initialized with preventDefault prop', () => {
```
Comment (miguelpinto98) [inline packages/ui-components/src/components/tree-item/test/tree-item.spec.tsx:80]: 🔝

**Pull request merged**